pkg/rest/apisurface.go

```diff
@@ -92,11 +92,22 @@ func (s *APISurface) ProvisionHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	status := http.StatusOK
+	// MUST be returned if the Service Instance was provisioned
+	// as a result of this request and Not async
+	status := http.StatusCreated
+
+	// MUST be returned if the Service Instance provisioning is in progress.
 	if response.Async {
 		status = http.StatusAccepted
 	}
 
+	if response.AlreadyCreated {
+		// MUST be returned if the Service Instance already exists,
+		// is fully provisioned, and the requested parameters
```

Comment:
> ... and the requested parameters are identical to the existing Service Instance

Should we check that as well here, or maybe in another PR?

Reply:
We currently have no state, the business logic would have to do that check as they should have the record of the already completed service instance.

I would love for the library to do this but it would have to store state, or at least expose a way for the broker implementer to define how the library should save the state.

I believe that this should be outside the scope of this PR if this library would prefer this behavior.